src/Container/ContainerLogStep.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Container;
+
+final class ContainerLogStep
```

Comment:
Could you explain what this class is used for? Preferably as a comment in the code, so that we don't forget 5 months from now :)

Reply:
I will make this all more clear in a bit. Still working on the implementation a tad.